From de90f477481de4bb2d8b7db27f58d2da71e863d8 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 09 Nov 2021 14:22:09 +0200
Subject: [PATCH] [SW-0] gaudi2/greco: pll: fix reset func semantics

The hl_pll_reset() function in Gaudi2 and Greco have a reversed
semantics:
true - means unreset
false - means reset.
Reverse the semantics to correspond to the function name and update the
callers to work with new semantics.

Change-Id: Ide90a32a18ad68e338a6254802de69185d2fbb35
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/common/pll.c b/arch/gaudi2/common/pll.c
index f906a78..6775eca 100644
--- a/arch/gaudi2/common/pll.c
+++ b/arch/gaudi2/common/pll.c
@@ -59,10 +59,10 @@
 {
 	u32 reg = regmap_readl(cfg->base + GAUDI2_PLL_CFG);
 
-	if (true == reset)
-		reg |= GAUDI2_PLL_CFG_EN;
-	else
+	if (reset)
 		reg &= ~GAUDI2_PLL_CFG_EN;
+	else
+		reg |= GAUDI2_PLL_CFG_EN;
 
 	regmap_writel(reg, cfg->base + GAUDI2_PLL_CFG);
 }
@@ -82,7 +82,7 @@
 	u32 reg = 0;
 
 	/* put the PLL into reset by writing 0 to CTRL_CFG.EN and CTRL_CFG.FOUTPOSTDIVEN */
-	hl_pll_reset(cfg, false);
+	hl_pll_reset(cfg, true);
 
 	/* FBDIV - bit 0
 	 * POSTDIV1 - bit 16
@@ -99,7 +99,7 @@
 	udelay(GAUDI2_PLL_MAX_WAIT_TIME);
 
 	/* enable the PLL by writing a value of 1 to CTRL_CFG.EN and CTRL_CFG.FOUTPOSTDIVEN */
-	hl_pll_reset(cfg, true);
+	hl_pll_reset(cfg, false);
 }
 
 void hl_pll_ack_int(struct hl_pll_cfg *cfg) { }
diff --git a/arch/greco/arc/pll.c b/arch/greco/arc/pll.c
index 2f2fbe5..e88522b 100644
--- a/arch/greco/arc/pll.c
+++ b/arch/greco/arc/pll.c
@@ -60,10 +60,10 @@
 {
 	u32 reg = regmap_readl(cfg->base + GRECO_PLL_CFG);
 
-	if (true == reset)
-		reg |= GRECO_PLL_CFG_EN;
-	else
+	if (reset)
 		reg &= ~GRECO_PLL_CFG_EN;
+	else
+		reg |= GRECO_PLL_CFG_EN;
 
 	regmap_writel(reg, cfg->base + GRECO_PLL_CFG);
 }
@@ -83,7 +83,7 @@
 	u32 reg = 0;
 
 	/* put the PLL into reset by writing 0 to CTRL_CFG.EN */
-	hl_pll_reset(cfg, false);
+	hl_pll_reset(cfg, true);
 
 	/* FBDIV - bit 0
 	 * POSTDIV1 - bit 16
@@ -100,7 +100,7 @@
 	udelay(GRECO_PLL_MAX_WAIT_TIME);
 
 	/* enable the PLL by writing a value of 1 to CTRL_CFG.EN*/
-	hl_pll_reset(cfg, true);
+	hl_pll_reset(cfg, false);
 }
 
 void hl_pll_ack_int(struct hl_pll_cfg *cfg) { }
